tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/MatterProvision.cs

```diff
@@ -164,64 +193,120 @@ public bool CanCreateMatter(Client client)
 
             var searchResultsVM = await matterRepositoy.GetMattersAsync(searchRequestVM);
             if (searchResultsVM.TotalRows > 0)
-            {
-                IList<MatterData> matterDataList = new List<MatterData>();
+            {                
+
+                dynamic matterDataList = new List<dynamic>();
                 IEnumerable<IDictionary<string, object>> searchResults = searchResultsVM.SearchResults;
+                var searchSchema = configuration.GetSection("Search");
+                string key1 = configuration.GetSection("Search").GetSection("ManagedPropertyMatterName").Value.ToString();
                 foreach (var searchResult in searchResults)
                 {
-                    MatterData matterData = new MatterData();
+                    dynamic matterData = new ExpandoObject();                    

```

Comment:
Instead of concrete MattertData object, dynamic object of type Expando object has been created to cater the needs of both matter center and project center